### PR TITLE
Moving corporate actions to Quant::Framework

### PR DESCRIPTION
**Pull request by mm-binary**

### CODE REVIEW

File: lib/Quant/Framework/CorporateAction.pm [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                    description => "Acquisition",
+                    effective_date => "15-Jul-15",
+                    flag => "N", #N means new action, U means updated action, D means cancelled action
+                }});
+ $corp->save();
+
+=cut
+
+use Moose;
+extends 'Quant::Framework::Utils::MarketData';    #we keep this as its not related to CouchDB
```
Comment (jy-binary) [inline lib/Quant/Framework/CorporateAction.pm]: remove the mention of Couchdb

File: lib/Quant/Framework/Utils/Test.pm [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+use Data::Chronicle::Writer;
+use Data::Chronicle::Reader;
+use Data::Chronicle::Mock;
+
+=head2 create doc()
+
+    Create a new document in the test database
+
+    params:
+    $yaml_couch_db  => The name of the entity in the YAML file (eg. promo_code)
```
Comment (jy-binary) [inline lib/Quant/Framework/Utils/Test.pm]: remove the mention of Couchdb

File: lib/Quant/Framework/Utils/Test.pm [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    $data_mod       => hasref of modifictions required to the data (optional)
+
+=cut
+
+sub create_doc {
+    my ($yaml_couch_db, $data_mod) = @_;
```
Comment (jy-binary) [inline lib/Quant/Framework/Utils/Test.pm]: remove the mention of Couchdb

File: lib/Quant/Framework/Utils/Test.pm [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    my $data    = $fixture->{$yaml_couch_db}{data};
+
+    die "Invalid yaml db name: $yaml_couch_db" if not defined $data;
+
+    # modify data?
+    for (keys %$data_mod) {
+        $data->{$_} = $data_mod->{$_};
+    }
+
+    # use class to create the Couch doc
```
Comment (jy-binary) [inline lib/Quant/Framework/Utils/Test.pm]: remove the mention of Couchdb

**Comment by shahrizal-shaari** [comment]
I looked at .travis.yml, I am not sure if we really need all those 4 versions of Perl?

File: lib/Quant/Framework/CorporateAction.pm (hunk 1 above)
Comment (mm-binary) [inline lib/Quant/Framework/CorporateAction.pm]: Fixed.

File: lib/Quant/Framework/Utils/Test.pm (hunk 2 above)
Comment (mm-binary) [inline lib/Quant/Framework/Utils/Test.pm]: Done.

File: lib/Quant/Framework/Utils/Test.pm (hunk 3 above)
Comment (mm-binary) [inline lib/Quant/Framework/Utils/Test.pm]: Done.

File: lib/Quant/Framework/Utils/Test.pm (hunk 4 above)
Comment (mm-binary) [inline lib/Quant/Framework/Utils/Test.pm]: Done.

**Comment by mm-binary** [comment]
As this is a CPAN module, it won't use company's account resources on Travis. As a result we are testing it against different perl versions just to make sure it wont break when in future we upgrade to newer versions.

**Pull request merged**